app/api/files/files.ts

```diff
@@ -0,0 +1,31 @@
+import { deleteUploadedFiles } from 'api/files/filesystem';
+import connections from 'api/relationships';
+import { search } from 'api/search';
+
+import model from './filesModel';
+import { validateFile } from '../../shared/types/fileSchema';
+import { FileType } from '../../shared/types/fileType';
+
+export const files = {
+  async save(file: FileType) {
+    const savedFile = await model.save(await validateFile(file));
+    await search.indexEntities({ sharedId: file.entity }, '+fullText');
+    return savedFile;
+  },
+
+  get: model.get.bind(model),
+
+  async delete(query: any = {}) {
+    const toDeleteFiles: FileType[] = (await model.get(query)) || { filename: '' };
```

Comment:
@joan Why the default is an object when get returns an array? Also, since it always returns an array don't think the default will be ever used.

Reply:
@konzz do you mean the query default ? it is a query object, not a file object, it can not be an array.

Reply:
No, I mean the "(await model.get(query)) || { filename: '' }" the fallback to be correct.

Reply:
yeah, the fallback is not needed anymore since i changed it to accept a query instead of an id, this is solved.